src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs

```diff
@@ -2167,6 +2167,210 @@ protected override void BeginProcessing()
+                        {
+                            int lastError = Marshal.GetLastWin32Error();
+                            Win32Exception exception = new Win32Exception(lastError);
+                            WriteObject(exception);
+                            WriteNonTerminatingError(
+                                service,
+                                ServiceComputerName,
+                                exception,
+                                "ComputerAccessDenied",
+                                ServiceResources.ComputerAccessDenied,
+                                ErrorCategory.PermissionDenied);
+                            continue;
+                        }
+                        hService = NativeMethods.OpenServiceW(
+                            hScManager,
+                            Name,
+                            NativeMethods.SERVICE_DELETE
+                            );
+                        if (IntPtr.Zero == hService)
+                        {
+                            int lastError = Marshal.GetLastWin32Error();
+                            Win32Exception exception = new Win32Exception(lastError);
+                            WriteNonTerminatingError(
+                                service,
+                                exception,
+                                "CouldNotRemoveService",
+                                ServiceResources.CouldNotSetService,
+                                ErrorCategory.PermissionDenied);
+                            continue;
+                        }
+
+                        bool status = NativeMethods.DeleteService(hService);
+
+                        if (!status)
+                        {
+                            int lastError = Marshal.GetLastWin32Error();
+                            Win32Exception exception = new Win32Exception(lastError);
+                            WriteNonTerminatingError(
+                                service,
+                                exception,
+                                "CouldNotRemoveService",
+                                ServiceResources.CouldNotRemoveService,
+                                ErrorCategory.PermissionDenied);
+                        }
+                    }
+                    finally
+                    {
+                        if (IntPtr.Zero != hService)
+                        {
+                            bool succeeded = NativeMethods.CloseServiceHandle(hService);
+                            if (!succeeded)
+                            {
+                                int lastError = Marshal.GetLastWin32Error();
+                                Win32Exception exception = new Win32Exception(lastError);
+                                WriteNonTerminatingError(
+                                    service,
+                                    exception,
+                                    "CouldNotRemoveService",
```

Comment:
This doesn't seem to the right error.  According to the docs, `CloseServiceHandle` only fails if it's an invalid handle which is probably never going to the be case here.  Anyways, even if this fails, it would seem that the service was still deleted.  Maybe this should just be a diagnostic assert?